src/inet/routing/leach/Leach.cc

```diff
@@ -0,0 +1,450 @@
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+//
+
+#include "inet/common/IProtocolRegistrationListener.h"
+#include "inet/common/ModuleAccess.h"
+#include "inet/common/ProtocolTag_m.h"
+#include "inet/linklayer/common/InterfaceTag_m.h"
+#include "inet/networklayer/common/L3AddressTag_m.h"
+#include "inet/physicallayer/wireless/common/contract/packetlevel/SignalTag_m.h"
+#include "inet/networklayer/common/L3AddressResolver.h"
+#include "inet/common/lifecycle/LifecycleOperation.h"
+#include "inet/common/lifecycle/ModuleOperations.h"
+#include "inet/common/lifecycle/NodeStatus.h"
+#include "inet/common/lifecycle/LifecycleController.h"
+#include "inet/routing/leach/Leach.h"
+
+namespace inet {
+namespace leach {
+
+Register_Enum(inet::leach::Leach, (Leach::ch, Leach::nch));
+Define_Module(Leach);
+
+Leach::Leach() {
+
+}
+
+Leach::~Leach() {
+    cancelAndDelete(event);
+    stop();
+}
+
+void Leach::initialize(int stage) {
+    RoutingProtocolBase::initialize(stage);
+
+    //reads from omnetpp.ini
+    if (stage == INITSTAGE_LOCAL) {
+        sequencenumber = 0;
+        host = getContainingNode(this);
+        interfaceTable.reference(this, "interfaceTableModule", true);
+
+        clusterHeadPercentage = par("clusterHeadPercentage");
+        numNodes = par("numNodes");
+
+        roundDuration = dblrand(0) * 10;
+        TDMADelayCounter = 1;
+        event = new cMessage("event");
+        roundDuration = dblrand(0) * 10;
+        wasCH = false;
+
```

Comment:
The following vectors are rather suspicious to me. you are reading the numNodes parameter, then you create some vectors with this size and then throw away those vectors. Note that these are local variables on the stack that has nothing to do to the similarly named member variables in the class. You are not initialize those variables here, you are just hiding them. As you are dynamically fill those vectors, you don't have to pre-allocate those vectors anyway. Just delete these lines and the `numNodes` parameter.